msgpack/encode.ts

```diff
@@ -0,0 +1,251 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { BytesList } from "../bytes/bytes_list.ts";
+
+export type EncodeType =
```

Comment:
How about `ValueType` instead? `EncodeType` sounds a bit strange to me because this is related to both `encode` and `decode`.

Also in `std/json` module, the similar type is called `JsonValue` https://github.com/denoland/deno_std/blob/3eeaa6324fd502850c9ad3df65c9e43b3dc1e40f/json/common.ts#L5